File: src/main/java/um/haberes/rest/model/LegajoCargoClaseImputacion.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class LegajoCargoClaseImputacion extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 4462866740002625960L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long legajoCargoClaseImputacionId;

	private Long legajoId;
	private Integer anho = 0;
	private Integer mes = 0;
	private Integer dependenciaId;
	private Integer facultadId;
	private Integer geograficaId;
	private Long cargoClaseId;
	private BigDecimal cuentaSueldos;
	private BigDecimal basico = BigDecimal.ZERO;
	private BigDecimal antiguedad = BigDecimal.ZERO;
	private BigDecimal cuentaAportes;

}

File: src/main/java/um/haberes/core/kotlin/model/LegajoCargoClaseImputacion.kt
package um.haberes.core.kotlin.model

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table
data class LegajoCargoClaseImputacion(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var legajoCargoClaseImputacionId: Long? = null,

    var legajoId: Long? = null,
    var anho: Int = 0,
    var mes: Int = 0,
    var dependenciaId: Int? = null,
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var cargoClaseId: Long? = null,
    var cuentaSueldos: BigDecimal? = null,
    var basico: BigDecimal = BigDecimal.ZERO,
    var antiguedad: BigDecimal = BigDecimal.ZERO,
    var cuentaAportes: BigDecimal? = null

) : Auditable()
